src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -97,6 +105,22 @@ def _generate_redis_response(self, response, simulation_id, is_successful, comma
                 })
         self.publish_json(response_channel, response_json)
 
+    def _area_map_callback(self, payload: Dict) -> None:
+        """Triggers the calculation of area uuid and name map and publish it
+        back to a redis response channel"""
+        area_map = self._area_uuid_name_map_wrapper(self._simulation.area)
+        response_channel = f"external-myco/{d3a.constants.CONFIGURATION_ID}/area-map/response/"
+        response_dict = {"area_map": area_map, "event": "area_map"}
+        self.publish_json(response_channel, response_dict)
+
+    @classmethod
+    def _area_uuid_name_map_wrapper(cls, area: "Area", area_map: Dict = {}) -> Dict:
+        """Recursive method to populate area uuid and name map for area object"""
```

Comment:
Better avoid using mutable default arguments
```suggestion
    def _area_uuid_name_map_wrapper(cls, area: "Area", area_mapping: Optional[dict] = None) -> Dict:
        """Recursive method to populate area uuid and name map for area object"""
        area_mapping = area_mapping or {}
```